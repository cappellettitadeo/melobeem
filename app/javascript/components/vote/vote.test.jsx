import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Vote from "./vote";

const mockProps = {
  location: {
    propsSearch: {
      id: `111`,
      name: `Julia Mur`,
      avatar: `some/url/avatar.png`,
      likes: 1000,
      parentName: `John Depp`,
      rank: 2,
    },
  },
  match: {
    params: `111`,
  },
};

describe("Components", () => {
  describe("Vote", () => {
    it("should render Vote component", () => {
      const component = renderer
        .create(
          <Router>
            <Vote location={mockProps.location} match={mockProps.match} />
          </Router>
        )
        .toJSON();

      expect(component).toMatchSnapshot();
    });
  });
});
