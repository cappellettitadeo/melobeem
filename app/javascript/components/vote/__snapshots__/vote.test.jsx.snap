// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Vote should render Vote component 1`] = `
Array [
  <div
    className="header-user"
  >
    <div
      className="header-user__item"
    >
      <div
        className="header-user__item__text text-tiny text-grey"
      >
        Votes
      </div>
      <div
        className="header-user__item__number headline--medium text-pink"
      >
        1
      </div>
    </div>
    <div
      className="half-circle header-user__level"
    >
      <div
        className="header-user__level__text text-grey text-tiny"
      >
        Level 1 
        <span
          className="text-tiny"
        >
          (1/5)
        </span>
      </div>
      <test-file-stub />
    </div>
    <div
      className="header-user__item"
    >
      <div
        className="header-user__item__text text-tiny text-grey"
      >
        Rank
      </div>
      <div
        className="header-user__item__number headline--medium text-pink"
      >
        1,789
      </div>
    </div>
  </div>,
  <div
    className="vote"
  >
    <div
      className="vote-item"
    >
      <div
        className="vote-item__img"
      >
        <test-file-stub
          className="vote-item__img--free"
        />
      </div>
      <div
        className="vote-item__text"
      >
        1 Vote
      </div>
      <div
        className="vote-item__button vote-item__button--orange"
      >
        00:29
      </div>
    </div>
    <div
      className="vote-item"
    >
      <div
        className="vote-item__img"
      >
        <test-file-stub
          className="vote-item__img--small"
        />
      </div>
      <div
        className="vote-item__text"
      >
        10 Votes
      </div>
      <div
        className="vote-item__button"
      >
        £ 10
      </div>
    </div>
    <div
      className="vote-item"
    >
      <div
        className="vote-item__img"
      >
        <test-file-stub
          className="vote-item__img--medium"
        />
      </div>
      <div
        className="vote-item__text"
      >
        20 Votes
      </div>
      <div
        className="vote-item__button"
      >
        £ 20
      </div>
    </div>
    <div
      className="vote-item"
    >
      <div
        className="vote-item__img"
      >
        <test-file-stub
          className="vote-item__img--big"
        />
      </div>
      <div
        className="vote-item__text"
      >
        50 Votes
      </div>
      <div
        className="vote-item__button"
      >
        £ 50
      </div>
    </div>
  </div>,
]
`;
